CMakeLists.txt

```diff
@@ -69,6 +69,24 @@ option(FMT_TEST "Generate the test target." ${FMT_MASTER_PROJECT})
 option(FMT_FUZZ "Generate the fuzz target." OFF)
 option(FMT_CUDA_TEST "Generate the cuda-test target." OFF)
 option(FMT_OS "Include core requiring OS (Windows/Posix) " ON)
+option(FMT_MODULE "Build a module instead of a traditional library." OFF)
+
+set(FMT_CAN_MODULE OFF)
+if (CMAKE_CXX_STANDARD GREATER_EQUAL 20)
+  if (MSVC AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.29.30035)) # msvc 16.10-pre3
+    set(FMT_CAN_MODULE ON)
+  endif ()
+endif ()
+if (FMT_CAN_MODULE)
+  if (MSVC)
+    set(FMT_BMI "${CMAKE_BINARY_DIR}/fmt.ifc")
+    set(BMI_BUILD_FLAGS /interface /ifcOutput ${FMT_BMI})
+    set(BMI_USE_FLAGS /reference "\"fmt=${FMT_BMI}\"")
+    endif ()
```

Comment:
Yeah. That's also about to change. My current private code is different and I thought about generating these variables by functions that contain all the compiler-specific knowledge on how to create the required compiler options and syntax from a name and a location given as function arguments. This function can then be used to both create a module when the user consumes {fmt} that way and also in the tests where the library must be compiled in both modes at the same time: as a traditional library _and_ a module (only for testing purposes) with a name different from the traditional library. This way all existing functional tests remain as they are and can reach into implementation details as they please. The additional tests for module-capable compilers will not interfere. They don't need to (and also can't) test the implementation aspect again but rather look at the public, user-facing API, possible regressions reagarding unwittingly exposing implementation details like `namespace detail` and contents of it, and (at least in the beginning) the salient properties of modules and their compiler-specific implementations.

Do you have an opinion about this?

Reply:
I think this function should take a target name and set the required properties instead of setting a bunch of variables, something like:

```cmake 
function(enable_modules name)
  if (MSVC)
    target_compile_options(${name}
      PRIVATE /interface /ifcOutput ${CMAKE_CURRENT_BINARY_DIR}
      INTERFACE /reference ${name}=${CMAKE_CURRENT_BINARY_DIR}/${name}.ifc)
    ...
  endif ()
endfunction()
```
Note that `${CMAKE_CURRENT_BINARY_DIR}` is probably not needed because that will be the current directory anyway.

Reply:
Very good idea! You are definitely more fluent in CMake than I am.

Reply:
This seemingly unnecessary `${CMAKE_CURRENT_BINARY_DIR}` is required to generate correct compiler options. Checked with both the ninja and the vcproj generator.